src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (userFromDbOptional.isPresent() && userFromDbOptional
+                .get().getPassword()
+                .equals(HashUtil.hashPassword(password, userFromDbOptional.get().getSalt()))) {
+            return userFromDbOptional.get();
+        }
+        throw new AuthenticationException("Can't authenticate user");
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (userFromDbOptional.isEmpty()
+                && !password.isEmpty()) {
```

Comment:
```suggestion
        if (userFromDbOptional.isEmpty()) {
```

we will add password validation later